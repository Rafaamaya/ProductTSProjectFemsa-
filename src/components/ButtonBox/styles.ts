import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 20,
  },
  containerButton: {
    backgroundColor: '#334FFA',
    borderRadius: 10,
    height: 50,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 6,
  },
  textButton: {
    fontSize: 16,
    fontWeight: '800',
    fontFamily: 'Avenir',
    color: '#FFFFFF',
  },
});
