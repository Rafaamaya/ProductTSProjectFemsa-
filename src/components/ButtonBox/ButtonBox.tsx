/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import styles from './styles';

interface Props {
  show: boolean;
  onPressLosers: () => void;
  onPressWinners: () => void;
  onPressAll: () => void;
}

const ButtonBox = ({
  show,
  onPressLosers,
  onPressWinners,
  onPressAll,
}: Props) => {

  return (
    <>
      {show ? (
        <View style={styles.container}>
          <TouchableOpacity onPress={onPressAll} style={styles.containerButton}>
            <Text style={styles.textButton}>Todos</Text>
          </TouchableOpacity>
        </View>
      ) : (
        <View style={{flexDirection: 'row', marginTop: 20}}>
          <TouchableOpacity
            onPress={onPressWinners}
            style={styles.containerButton}>
            <Text style={styles.textButton}>Ganados</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.containerButton}
            onPress={onPressLosers}>
            <Text style={styles.textButton}>Canjeados</Text>
          </TouchableOpacity>
        </View>
      )}
    </>
  );
};

export default ButtonBox;
