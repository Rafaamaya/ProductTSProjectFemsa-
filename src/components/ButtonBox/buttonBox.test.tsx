import React from 'react';
import renderer from 'react-test-renderer';
import ButtonBox from './ButtonBox';

const show = true;
const onPressLosers = function (): void {
  throw new Error('Function not implemented.');
};
const onPressWinners = function (): void {
  throw new Error('Function not implemented.');
};
const onPressAll = function (): void {
  throw new Error('Function not implemented.');
};

describe('Pruebas en <ButtonBox/>', () => {
  test('debe de hacer match con el snapchot', () => {
    const tree = renderer
      .create(
        <ButtonBox
          show={show}
          onPressLosers={onPressLosers}
          onPressWinners={onPressWinners}
          onPressAll={onPressAll}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
