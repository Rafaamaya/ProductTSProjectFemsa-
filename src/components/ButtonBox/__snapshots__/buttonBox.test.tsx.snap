// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ButtonBox/> debe de hacer match con el snapchot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 20,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#334FFA",
        "borderRadius": 10,
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "marginRight": 6,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir",
          "fontSize": 16,
          "fontWeight": "800",
        }
      }
    >
      Todos
    </Text>
  </View>
</View>
`;
